aiida_mlip/calculations/train.py

```diff
@@ -0,0 +1,170 @@
+
+
+def validate_inputs(
+    inputs: dict, port_namespace: aiida.engine.processes.ports.PortNamespace
+):
+    """
+    Check if the inputs are valid.
+
+    Parameters
+    ----------
+    inputs : dict
+        The inputs dictionary.
+
+    port_namespace : `aiida.engine.processes.ports.PortNamespace`
+        An instance of aiida's `PortNameSpace`.
+
+    Raises
+    ------
+    InputValidationError
+        Error message if validation fails, None otherwise.
+    """
+    if "mlip_config" in port_namespace:
+        if "mlip_config" not in inputs:
+            raise InputValidationError("No config file given")
+        mlip_dict = inputs.mlip_config.as_dictionary()
+        required_keys = ("train_file", "valid_file", "test_file", "name")
+        for key in required_keys:
+            if key not in mlip_dict:
+                raise InputValidationError(f"Mandatory key {key} not in config file")
+            # Check if the keys actually correspond to a path except name which is
+            # just the name to use for the output files
+            if key != "name":
+                if not Path(key).exists():
+                    raise InputValidationError(f"Path given for {key} does not exist")
+
+
+class Train(CalcJob):  # numpydoc ignore=PR01
+    """
+    Calcjob implementation to train mlips.
+
+    Attributes
+    ----------
+    DEFAULT_OUTPUT_FILE : str
+        Default stdout file name.
+    DEFAULT_INPUT_FILE : str
+        Default input file name.
+    LOG_FILE : str
+        Default log file name.
+
+    Methods
+    -------
+    define(spec: CalcJobProcessSpec) -> None:
+        Define the process specification, its inputs, outputs and exit codes.
+    validate_inputs(value: dict, port_namespace: PortNamespace) -> Optional[str]:
+        Check if the inputs are valid.
+    prepare_for_submission(folder: Folder) -> CalcInfo:
+        Create the input files for the `CalcJob`.
+    """
+
+    DEFAULT_OUTPUT_FILE = "aiida-stdout.txt"
+    DEFAULT_INPUT_FILE = "aiida.xyz"
+    LOG_FILE = "aiida.log"
+
+    @classmethod
+    def define(cls, spec: CalcJobProcessSpec) -> None:
+        """
+        Define the process specification, its inputs, outputs and exit codes.
+
+        Parameters
+        ----------
+        spec : `aiida.engine.CalcJobProcessSpec`
+            The calculation job process spec to define.
+        """
+        super().define(spec)
+        spec.inputs.validator = validate_inputs
+        # Define inputs
+        spec.input(
+            "mlip_config",
+            valid_type=JanusConfigfile,
+            required=False,
+            help="Mlip architecture to use for calculation, defaults to mace",
+        )
+        spec.output("model", valid_type=ModelData)
+        spec.output("compiled_model", valid_type=SinglefileData)
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
```

Comment:
```suggestion
            help="The `results_dict` output node of the training.",
```
?

(I'm not completely sure what goes into the output node, so there may well be a better description/it may be fine as is)